tarfs/tarfs_test.go

```diff
@@ -0,0 +1,402 @@
+// Most of the tests are stolen from the zipfs implementation
+package tarfs
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"syscall"
+	"testing"
+
+	"github.com/spf13/afero"
+)
+
+var files = []struct {
+	name        string
+	exists      bool
+	isdir       bool
+	size        int64
+	content     string
+	contentAt4k string
+}{
+	{"/", true, true, 0, "", ""},
+	{"/sub", true, true, 0, "", ""},
+	{"/sub/testDir2", true, true, 0, "", ""},
+	{"/sub/testDir2/testFile", true, false, 8192, "cccccccc", "ccccdddd"},
+	{"/testFile", true, false, 8192, "aaaaaaaa", "aaaabbbb"},
+	{"/testDir1/testFile", true, false, 8192, "bbbbbbbb", "bbbbcccc"},
+
+	{"/nonExisting", false, false, 0, "", ""},
+}
+
+var dirs = []struct {
+	name     string
+	children []string
+}{
+	{"/", []string{"sub", "testDir1", "testFile"}},
+	{"/sub", []string{"testDir2"}},
+	{"/sub/testDir2", []string{"testFile"}},
+	{"/testDir1", []string{"testFile"}},
+}
+
+var afs *afero.Afero
+
+func TestMain(m *testing.M) {
+	tf, err := os.Open("testdata/t.tar")
```

Comment:
You probably have to fix this path to work for Windows in order to make appveyor pass.

Reply:
Hi, thanks! I think that this is OK, the panic is how I handle the "pseudo-root" entry internally, as I use a literal slash, and to be honest I didn't think on Windows systems. I set up an AppVeyor account and will debug the issue.